src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java

```diff
@@ -0,0 +1,135 @@
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.SlotWidget;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
+import net.minecraft.world.storage.loot.LootContext;
+import net.minecraft.world.storage.loot.LootTable;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.ItemHandlerHelper;
+import net.minecraftforge.items.ItemStackHandler;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+
+public class MetaTileEntityFisher extends MetaTileEntity {
+
+    private IEnergyContainer energyContainer;
+    private static final int inventorySize = 25;
+    private static final long fishingTicks = 400L;
+    private static final long energyAmountPerFish = 128L;
+
+    public MetaTileEntityFisher(ResourceLocation metaTileEntityId){
+        super(metaTileEntityId);
+        this.energyContainer = new EnergyContainerHandler(this,
+            1024L, 32L, 1L, 0L, 0L);
+        initializeInventory();
+    }
+
+    public MetaTileEntity createMetaTileEntity(MetaTileEntityHolder holder) {
+        return new MetaTileEntityFisher(metaTileEntityId);
+    }
+
+    protected ModularUI createUI(EntityPlayer entityPlayer) {
+        int rowSize = (int) Math.sqrt(inventorySize);
+
+        ModularUI.Builder builder = ModularUI.builder(GuiTextures.BACKGROUND, 176,
+            18 + 18 * rowSize + 94)
+            .label(10, 5, getMetaFullName())
+            .widget(new SlotWidget(importItems, 0, 18, 18, true, true)
+                .setBackgroundTexture(GuiTextures.SLOT, GuiTextures.IN_SLOT_OVERLAY));
+
+        for (int y = 0; y < rowSize; y++) {
+            for (int x = 0; x < rowSize; x++) {
+                int index = y * rowSize + x;
+                builder.widget(new SlotWidget(exportItems, index, 89 - rowSize * 9 + x * 18, 18 + y * 18, true, false)
+                    .setBackgroundTexture(GuiTextures.SLOT, GuiTextures.OUT_SLOT_OVERLAY));
+            }
+        }
+
+        builder.bindPlayerInventory(entityPlayer.inventory, GuiTextures.SLOT, 8, 18 + 18 * rowSize + 12);
+        return builder.build(getHolder(), entityPlayer);
+    }
+
+    public void update() {
+        super.update();
+        if (!getWorld().isRemote && energyContainer.getEnergyStored() >= energyAmountPerFish && getTimer() % fishingTicks == 0L &&
+            importItems.getStackInSlot(0).getCount() > 0) {
+            WorldServer world = (WorldServer) this.getWorld();
+            BlockPos waterCheckPos = getPos().down();
+            if (world.getBlockState(waterCheckPos).getBlock().getRegistryName().toString().equals("minecraft:water")) {
+                LootTable table = world.getLootTableManager().getLootTableFromLocation(new ResourceLocation("minecraft:gameplay/fishing"));
```

Comment:
Don't use string constants for resource names, there's LootTables containing resource locations of all vanilla loottables